command_line/dlsnxs2cbf.py

```diff
@@ -0,0 +1,220 @@
+    else:
+        Bx = f["/entry/instrument/detector/beam_center_x"][()]
+        By = f["/entry/instrument/detector/beam_center_y"][()]
+
+    result.append("###CBF: VERSION 1.5, CBFlib v0.7.8 - Eiger detectors")
+    result.append("")
+    result.append("data_%06d" % (nn + 1))
+    result.append("")
+    result.append(
+        """_array_data.header_convention "PILATUS_1.2"
+_array_data.header_contents
+;"""
+    )
+    result.append("# Detector: EIGER 2XE 16M S/N 160-0001 Diamond")
+    result.append("# %s" % f["/entry/start_time"][()])
+    result.append("# Pixel_size 75e-6 m x 75e-6 m")
+    result.append("# Silicon sensor, thickness 0.000450 m")
+    result.append("# Exposure_time %.5f s" % T)
+    result.append("# Exposure_period %.5f s" % T)
+    result.append("# Tau = 1e-9 s")
+    result.append("# Count_cutoff 65535 counts")
+    result.append("# Threshold_setting: 0 eV")
+    result.append("# Gain_setting: mid gain (vrf = -0.200)")
+    result.append("# N_excluded_pixels = 0")
+    result.append("# Excluded_pixels: badpix_mask.tif")
+    result.append("# Flat_field: (nil)")
+    result.append("# Wavelength %.5f A" % L)
+    result.append("# Detector_distance %.5f m" % (D / 1000.0))
+    result.append("# Beam_xy (%.2f, %.2f) pixels" % (Bx, By))
+    result.append("# Flux 0.000000")
+    result.append("# Filter_transmission %.3f" % A)
+    result.append("# Start_angle %.4f deg." % omega[nn])
+    result.append("# Angle_increment %.4f deg." % omega_increment[nn])
+    result.append("# Detector_2theta 0.0000 deg.")
+    result.append("# Polarization 0.990")
+    result.append("# Alpha 0.0000 deg.")
+    result.append("# Kappa 0.0000 deg.")
+    result.append("# Phi %.4f deg." % phi)
+    result.append("# Phi_increment 0.0000 deg.")
+    result.append("# Omega %.4f deg." % omega[nn])
+    result.append("# Omega_increment %.4f deg." % omega_increment[nn])
+    result.append("# Chi %.4f deg." % chi)
+    result.append("# Chi_increment 0.0000 deg.")
+    result.append("# Oscillation_axis X.CW")
+    result.append("# N_oscillations 1")
+    result.append(";")
+
+    return "\n".join(result)
+
+
+def pack(data):
+    from cbflib_adaptbx import compress
+
+    return compress(data)
+
+
+def make_cbf(in_name, template):
+    f = h5py.File(in_name, "r")
+    depends_on(f)
+
+    global datasets
+
+    import binascii
+
+    start_tag = binascii.unhexlify("0c1a04d5")
+
+    for j in range(len(f["/entry/sample/transformations/omega"][()])):
+        block = 1 + (j // 1000)
+        i = j % 1000
+        header = print_cbf_header(f, j)
+        depth, height, width = f["/entry/data/data_%06d" % block].shape
+        import numpy
+
+        data = flex.int(numpy.int32(f["/entry/data/data_%06d" % block][i]))
+        good = data.as_1d() < 65535
+        data.as_1d().set_selected(~good, -2)
+
+        # set the tile join regions to -1 - MOSFLM cares about this apparently
+        mask = get_mask(width, height)
```

Comment:
Could you just cache the result of mask locally in this function, and only call `get_mask()` `if mask is None` rather than messing around with the global variable `_mask`?